crawl-ref/source/main.cc

```diff
@@ -447,6 +447,12 @@ NORETURN static void _launch_game()
 
     _prep_input();
 
+    if (you.species == SP_FAERIE_DRAGON && you.num_turns == 0
```

Comment:
These conditionals should be moved into `float_once()`, as the faerie dragon check is currently duplicated. It should also be given a more descriptive name; consider making it static to main.cc and calling it `_faerie_dragon_begin_game_flying()`. You might want to consider using `no_messages` to silence `float_player()`, if the player is supposed to be already flying when they begin the game.